src/Graphics/OpenGLContext/opengl_WrappedFunctions.h

```diff
@@ -0,0 +1,2629 @@
+#pragma once
+
+#include <thread>
+#include <mutex>
+#include <atomic>
+#include <condition_variable>
+#include <memory>
+#include <vector>
+#include "BlockingQueue.h"
+#include "GLFunctions.h"
+#include "opengl_Attributes.h"
+#include <algorithm>
+#include <functional>
+#include <unordered_map>
+
+
+#ifdef MUPENPLUSAPI
+#include <mupenplus/GLideN64_mupenplus.h>
+#else
+#include <Graphics/OpenGLContext/windows/WindowsWGL.h>
+#endif
+
+namespace opengl {
+
+	class OpenGlCommand
+	{
+	public:
+		void performCommandSingleThreaded(void)
+		{
+			commandToExecute();
+#ifdef GL_DEBUG
+			if(m_isGlCommand)
+			{
+				auto error = g_glGetError();
+				if (error != GL_NO_ERROR) {
+					std::stringstream errorString;
+					errorString << " OpenGL error: 0x" << std::hex << error << ", on function: " << m_functionName;
+					LOG(LOG_ERROR, errorString.str().c_str());
+					throw std::runtime_error(errorString.str().c_str());
+				}
+			}
+#endif
+		}
+
+		void performCommand(void) {
+			std::unique_lock<std::mutex> lock(m_condvarMutex);
+			performCommandSingleThreaded();
+			if (m_synced)
+			{
+#ifdef GL_DEBUG
+				if (m_logIfSynced) {
+					std::stringstream errorString;
+					errorString << " Executing synced: " << m_functionName;
+					LOG(LOG_ERROR, errorString.str().c_str());
+				}
+#endif
+				m_executed = true;
+				m_condition.notify_all();
+			}
+		}
+
+		void waitOnCommand(void) {
+			std::unique_lock<std::mutex> lock(m_condvarMutex);
+
+			if (m_synced && !m_executed) {
+				m_condition.wait(lock, [this]{return m_executed;});
+			}
+		}
+
+	protected:
+		OpenGlCommand (const bool& _synced, const bool& _logIfSynced, const std::string& _functionName,
```

Comment:
I was trying to allow the compiler to optimize. In hindsight, it probably makes no difference because the compiler will already do this for you.